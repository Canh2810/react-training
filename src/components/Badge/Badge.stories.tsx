import { StoryObj, Meta } from '@storybook/react'

import Badge from '.'
import { commonVariants } from '@types'

export default {
  title: 'Components/Badge',
  component: Badge,
} as Meta

type Story = StoryObj<typeof Badge>

export const Default: Story = {
  args: {
    title: 'Badge',
  },
}

export const Secondary: Story = {
  args: {
    title: 'Badge',
    variant: commonVariants.Secondary,
  },
}
