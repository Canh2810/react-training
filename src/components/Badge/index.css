.badge {
  padding: 4px 10px;
  border-radius: 6px;
  font-size: var(--xs-font);
  font-weight: var(--semi-bold);
  line-height: var(--xs-line-height);
}
