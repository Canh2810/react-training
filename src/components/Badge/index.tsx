// Themes
import { variants } from '@themes'

// Types
import { commonVariants } from '@types'

// Style
import './index.css'

export interface BadgeProps {
  variant?: commonVariants
  title: string
}

const Badge = ({ variant = commonVariants.Primary, title }: BadgeProps) => {
  const badgeStyle = variants[variant]
  return (
    <span className="badge" style={badgeStyle}>
      {title}
    </span>
  )
}

export default Badge
