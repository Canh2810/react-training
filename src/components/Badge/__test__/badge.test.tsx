import { render } from '@testing-library/react'
import Badge from '..'
import { commonVariants } from '@types'

describe('Badge component', () => {
  it('should render primary variant', () => {
    const { container } = render(<Badge title="Test Badge" />)
    expect(container).toMatchSnapshot()
  })
  it('should render secondary variant', () => {
    const { container } = render(
      <Badge variant={commonVariants.Secondary} title="Test Badge" />,
    )
    expect(container).toMatchSnapshot()
  })
})
