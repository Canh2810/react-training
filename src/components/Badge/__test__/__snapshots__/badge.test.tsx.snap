// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component should render primary variant 1`] = `
<div>
  <span
    class="badge"
    style="background-color: rgb(75, 107, 251); color: rgb(255, 255, 255);"
  >
    Test Badge
  </span>
</div>
`;

exports[`Badge component should render secondary variant 1`] = `
<div>
  <span
    class="badge"
    style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
  >
    Test Badge
  </span>
</div>
`;
